ops/pebble.py

```diff
@@ -1621,46 +1622,76 @@ def _request_raw(
                 # Will only happen on read error or if Pebble sends invalid JSON.
                 body: Dict[str, Any] = {}
                 message = f'{type(e2).__name__} - {e2}'
-            raise APIError(body, code, status, message)
+            raise APIError(body, code, status, message) from None
         except urllib.error.URLError as e:
-            raise ConnectionError(e.reason)
+            raise ConnectionError(e.reason) from e
 
         return response
 
     def get_system_info(self) -> SystemInfo:
-        """Get system info."""
+        """Get system info.
+
+        Raises:
+            ConnectionError: if pebble cannot be reached.
+            APIError: if an error occurred communicating with pebble.
```

Comment:
Similar to above, let's just document all these common exceptions on `pebble.Client` for now.